sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -142,19 +142,25 @@ object AggUtils {
 
     val distinctAttributes = normalizedNamedDistinctExpressions.map(_.toAttribute)
     val groupingAttributes = groupingExpressions.map(_.toAttribute)
+    val distinctFilterAttributes = functionsWithDistinct.flatMap(_.filterAttributes.toSeq)
 
     // 1. Create an Aggregate Operator for partial aggregations.
     val partialAggregate: SparkPlan = {
       val aggregateExpressions = functionsWithoutDistinct.map(_.copy(mode = Partial))
       val aggregateAttributes = aggregateExpressions.map(_.resultAttribute)
       // We will group by the original grouping expression, plus an additional expression for the
-      // DISTINCT column. For example, for AVG(DISTINCT value) GROUP BY key, the grouping
-      // expressions will be [key, value].
+      // DISTINCT column and the referred attributes in the FILTER clause associated with each
+      // aggregate function. For example,
+      // 1.for AVG(DISTINCT value) GROUP BY key, the grouping expressions will be [key, value].
+      // 2.for AVG (DISTINCT value) Filter (WHERE age > 20) GROUP BY key, the grouping expression
+      // will be [key, value, age].
```

Comment:
`AVG (DISTINCT value) Filter (WHERE age > 20)` this will be rewritten as `AVG (DISTINCT _gen_attr) Filter (WHERE _gen_attr is not null)`.  So here we should group by `key` and `_gen_attr`?

Reply:
Thanks for your remind.